### PR TITLE
Do not enforce network key presence for collators

### PR DESCRIPTION
**Pull request by alexggh**
Because of https://github.com/paritytech/polkadot-sdk/blob/299aacb56f4f11127b194d12692b00066e91ac92/cumulus/client/cli/src/lib.rs#L318 https://github.com/paritytech/polkadot-sdk/pull/3852 wrongly enforces that the network key is present for collators started with polkadot-parachain binary, instead of generating it on the fly.

That is not necessary and created some small friction for collators, since they have to pass `--unsafe-force-node-key-generation` or generate the key with the `generate-node-key` command.

This PR fixes that since the change was intended to apply only for relaychain authorithies.
Triggered by: https://matrix.to/#/!gTDxPnCbETlSLiqWfc:matrix.org/$Y9hLdT0z2TZCtskIcsIrD3kQm3XE5nga4b4LbGWsh-8?via=web3.foundation&via=matrix.org&via=parity.io

### CODE REVIEW

**Review by bkchr — CHANGES_REQUESTED** [review]
I mean it was quite clear from the beginning that this is required for every authority. I also don't really see why collators should not do the same. I mean yes, right now this is not required, but I think at some point we will also use authority discovery for parachains. Generally I see this as some kind of "hygiene". It adds some friction, but it is the same as for relay chain validator operators. 

TLDR: I would keep it this way and close this pr.

**Comment by alexggh** [comment]
> I mean it was quite clear from the beginning that this is required for every authority. I also don't really see why collators should not do the same. I mean yes, right now this is not required, but I think at some point we will also use authority discovery for parachains. Generally I see this as some kind of "hygiene". It adds some friction, but it is the same as for relay chain validator operators.

Indeed, this is a valid point I did not know we planned to use authority-discovery for parachains as well, strictly speaking we don't need to enforce this for parachains at this moment, but if we envision that will happen in the future I agree with you that this PR does not make much sense.

> TLDR: I would keep it this way and close this pr.

Given, I have just one complain datapoint for this since v1.11 and all the above I tend to agree, so closing the PR.
